app/utils/preloadReady.ts

```diff
@@ -0,0 +1,21 @@
+export default async function waitForPreloadReady(timeout = 5000, interval = 50): Promise<void> {
```

Comment:
Great! While it annoys me that this is necessary (as the preload script is supposed to be available as per the election docs) it seems like a good way to solve the race condition combined with the top level `await`.

I think it would be good to add docstring for this function explaining why it's there:
```suggestion
/**
 * This function is intended to be used by modules using `window.database` to interact with the database API in 
 * the preload script. This solves a race condition where this object is sometimes not available when the renderer 
 * script is executed. The problem has been observed on windows specifically with electron version 12 and 
 * election-builder version 23.
 * 
 * @returns A promise which resolves when `window.database` has been populated.
 * 
 * @example
 * await waitForPreloadReady();
 * ...
 */
export default async function waitForPreloadReady(timeout = 5000, interval = 50): Promise<void> {
```